src/main/java/io/confluent/idesidecar/websocket/proxy/ProxyContext.java

```diff
@@ -47,7 +55,51 @@ public ProxyContext withConnection(CCloudConnectionState connection) {
   }
 
   public String getConnectUrl() {
-    // TODO: I guess this won't work for private networks, we'll need something more sophisticated
+    String privateEndpoint = getMatchingPrivateEndpoint();
+    if (privateEndpoint != null) {
+      return transformToLanguageServiceUrl(privateEndpoint);
+    }
+
+    return buildPublicUrl();
+  }
+
+  private String getMatchingPrivateEndpoint() {
```

Comment:
What about using `Optional<String>` as the return value here, so that the method can return an empty optional if there's no matching private endpoint?

In general, it's good practice to use [Optional](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html)s because they offer stronger typing and prevents `NullPointerException`s.